src/components/Table/AutoDataTable/index.vue

```diff
@@ -482,17 +479,12 @@ export default {
       }
       this.popoverColumns.currentCols = columns
 
-      const _tableConfig = localStorage.getItem('tableConfig')
-        ? JSON.parse(localStorage.getItem('tableConfig'))
-        : {}
       let tableName = this.config.name || this.$route.name + '_' + newURL(url).pathname
       // 替换url中的uuid，避免同一个类型接口生成多个key，localStorage中的数据无法共用.
       tableName = replaceAllUUID(tableName)
 
-      _tableConfig[tableName] = {
-        'showColumns': columns
-      }
-      localStorage.setItem('tableConfig', JSON.stringify(_tableConfig))
+      this.objTableColumns.set(tableName, columns)
+
       this.filterShowColumns()
     },
     filterChange(filters) {
```

Comment:
No difference found. This is an existing component with some formatters implemented in it already which seem quite well-optimized. I'd say there's no need for further optimization at this point without more specific information about what needs to change or add functionality.